backend/typescript/services/implementations/stripeService.ts

```diff
@@ -0,0 +1,60 @@
+import dotenv from "dotenv";
+
+import Stripe from "stripe";
+import IStripeService from "../interfaces/stripeService";
+import logger from "../../utilities/logger";
+
+dotenv.config();
+
+const stripe = new Stripe(process.env.STRIPE_PRIVATE_KEY as string);
+const Logger = logger(__filename);
+const SUCCESS_URL = `http://localhost:5000/checkout-success`;
+const CANCEL_URL = `http://localhost:5000/checkout-cancel`;
```

Comment:
what's going on here? does stripe hit these endpoints upon a successful or unsuccessful transaction (like webhook events)?

if so, we've [completed work involving setting up stripe webhooks](https://www.notion.so/uwblueprintexecs/create-Stripe-webhooks-for-successful-transactions-b04a9056da104634977f4e0420637ad1?pvs=25) already.

if it's something else though, we don't want to hardcode our backend URL in this file here. we should pass in the URL from our environment variables

Reply:
not related to webhooks.

it's for a non-embedded checkout session. if you X out of it you go to the cancel url. if you go though with payment you go to the success url. Theyre required fields for checkout session, but since nothing exists yet we just put placeholders

that being said, we'd like some clarity on whether the checkout session should be embedded (injected into webpage) or non-embedded when more is finalized. We coded up non-embedded because otherwise we'd need a frontend component to consume this endpoint as well.

Reply:
I see, thanks for leaving this comment. Apologies, I should've provided clarity on this before you guys started this ticket — I hadn't considered this ticket would involve this stuff.

The checkout session should follow the non-embedded approach, as in users should be redirected to a Stripe-hosted checkout page. Then, after completing or cancelling the payment, they are redirected back to a checkout success or cancellation page respectively (as you have set up now).

This approach reduces much of our frontend workload and helps with security.

Just add a small change — the base url (`http://localhost:5000/`) should be passed in dynamically from the environment variables, and maybe add a short comment that those links you defined in `stripeService.ts` are to be used in the future for checkout success/cancellation pages.

Thanks for all the work with this ticket! Once this small change is done, I'll approve this PR and it can be merged! 🙏